lighthouse-core/test/audits/first-meaningful-paint-test.js

```diff
@@ -10,86 +10,22 @@ const Audit = require('../../audits/audit.js');
 
-function generateArtifactsWithTrace(trace) {
-  return Object.assign({
-    traces: {
-      [Audit.DEFAULT_PASS]: {traceEvents: Array.isArray(trace) ? trace : trace.traceEvents},
-    },
-    devtoolsLogs: {[Audit.DEFAULT_PASS]: []},
-  }, computedArtifacts);
-}
-
 /* eslint-env mocha */
 describe('Performance: first-meaningful-paint audit', () => {
-  const context = {options, settings: {throttlingMethod: 'provided'}};
-
-  describe('measures the pwa.rocks example correctly', () => {
-    let fmpResult;
-
-    it('processes a valid trace file', () => {
-      return FMPAudit.audit(generateArtifactsWithTrace(traceEvents), context).then(result => {
-        fmpResult = result;
-      }).catch(_ => {
-        assert.ok(false);
-      });
-    });
-
-    it('finds the expected fMP', () => {
-      assert.equal(fmpResult.displayValue, '1,100\xa0ms');
-      assert.equal(fmpResult.rawValue, 1099.523);
-    });
-
-    it('scores the fMP correctly', () => {
-      assert.equal(fmpResult.score, 0.99);
-    });
-  });
-
-  describe('finds correct FMP', () => {
-    it('if there was a tracingStartedInPage after the frame\'s navStart', () => {
-      const artifacts = generateArtifactsWithTrace(lateTracingStartedTrace);
-      return FMPAudit.audit(artifacts, context).then(result => {
-        assert.equal(result.displayValue, '530\xa0ms');
-        assert.equal(result.rawValue, 529.916);
-      });
-    });
-
-    it('if there was a tracingStartedInPage after the frame\'s navStart #2', () => {
-      const artifacts = generateArtifactsWithTrace(badNavStartTrace);
-      return FMPAudit.audit(artifacts, context).then(result => {
-        assert.equal(result.displayValue, '630\xa0ms');
-        assert.equal(result.rawValue, 632.424);
-      });
-    });
-
-    it('if it appears slightly before the fCP', () => {
-      return FMPAudit.audit(generateArtifactsWithTrace(preactTrace), context).then(result => {
-        assert.equal(result.displayValue, '880\xa0ms');
-        assert.equal(result.rawValue, 878.353);
-      });
-    });
-
-    it('from candidates if no defined FMP exists', () => {
-      return FMPAudit.audit(generateArtifactsWithTrace(noFMPtrace), context).then(result => {
-        assert.equal(result.displayValue, '4,460\xa0ms');
-        assert.equal(result.rawValue, 4460.928);
-      });
-    });
-  });
-
-  it('handles traces missing an FMP', () => {
-    return FMPAudit.audit(generateArtifactsWithTrace(noFCPtrace), context).then(result => {
-      assert.strictEqual(result.debugString, undefined);
-      assert.strictEqual(result.displayValue, '480\xa0ms');
-      assert.strictEqual(result.rawValue, 482.318);
-    });
+  it('computes FMP correctly for valid trace', async () => {
+    const artifacts = Object.assign({
+      traces: {[Audit.DEFAULT_PASS]: {traceEvents}},
+      devtoolsLogs: {[Audit.DEFAULT_PASS]: []},
+    }, computedArtifacts);
+    const context = {options, settings: {throttlingMethod: 'provided'}};
```

Comment:
maybe also a `simulated` test (and `devtools`? Even though it's currently the same execution path as this) so it checks that it can also successfully call through?

Reply:
done